build_tools/cmake/iree_run_module_test.cmake

```diff
@@ -145,7 +147,14 @@ function(iree_run_module_test)
       else()
         file(RELATIVE_PATH _OUTPUT_FILE_RELATIVE_PATH
           "${CMAKE_CURRENT_BINARY_DIR}" "${_OUTPUT_FILE_ABS_PATH}")
-        list(APPEND _RULE_RUNNER_ARGS "--expected_output=@${_OUTPUT_FILE_RELATIVE_PATH}")
+        # Android runs on device and pushs the file into a different path.
+        if(ANDROID)
+          cmake_path(GET _OUTPUT_FILE_ABS_PATH FILENAME _OUTPUT_FILE_BASE)
+          list(APPEND
+            _RULE_RUNNER_ARGS "--expected_output=@${_PACKAGE_PATH}/${_RULE_NAME}/${_OUTPUT_FILE_BASE}")
```

Comment:
I feel like we should extend the `TEST_INPUT_FILE_ARG` of `iree_native_test` to named arguments, so these logics can be put together. Can you file a bug and add TODO here to track this?

Reply:
https://github.com/iree-org/iree/issues/10744 filed.